searxng/searxng
# Simple theme svg2png

## What does this PR do?

In the master branch, convert (imagemagick) creates `searx/static/themes/simple/img/favicon.png` from `src/brand/searxng-wordmark.svg`. Some versions of convert have issues to render the .svg file.

This PR replaces the usage of convert by [sharp](https://github.com/lovell/sharp) (a standalone lib, it requires C++ to compile the library).

A custom grunt task is created because [grunt-task](https://www.npmjs.com/package/grunt-sharp) can't be installed (most probably because it hasn't been updated since a long time).

The "[build] /static" has been made using this change (the png file is smaller, it can be even smaller using [optipng](https://www.npmjs.com/package/optipng))

## Why is this change important?

Reliable build steps

## How to test this PR locally?

* `make themes.simple` or `npm run build`
* check `searx/static/themes/simple/img/favicon.png` is not modified (so `make static.build.commit` won't include the favicon at the build).
* check the favicon is rendered as expected.

## Author's checklist



## Related issues
